### PR TITLE
fix st_mode for MemoryPathIO and change build_list_string to use stat…

### PR DESCRIPTION
**Pull request by rsichny**
….filemode

### CODE REVIEW

**Comment by rsichny** [comment]
also changed `lambda` to `def` in asyncpathio to make flake8 happy

File: aioftp/pathio.py [hunk 1]
```diff
@@ -763,19 +766,20 @@ class Lister(AbstractAsyncLister):
         else:
 
             if node.type == "file":
-
```
Comment (pohmelie) [inline aioftp/pathio.py]: This empty line should be here. Since whole project written with some code convensions.

File: aioftp/pathio.py [hunk 2]
```diff
@@ -482,7 +484,8 @@ def worker(self):
             @with_timeout
             async def __aiter__(self):
 
-                f = lambda: path.glob("*")
+                def f():
+                    return path.glob("*")
```
Comment (pohmelie) [inline aioftp/pathio.py]: Also empty lines before and after `return`

**Comment by rsichny** [comment]
sorry, i'm a bit confused, sometimes you have empty lines in existing code, sometimes you don't... fixed.

**Comment by pohmelie** [comment]
@rsichny 
> sometimes you have empty lines in existing code, sometimes you don't...

Oh, I'm trying at my best :laughing:  If you find such situation please make issue/pull. Code style is important for me.

Pull is great!
The only thing I want to change is `from x import y`, since `stat` is not so long and `stat.y` is a bit cleaner.

Also, I don't remember why I use `append`, but we can just write
``` python
fields = (
    filemode(stats.st_mode),
    ...
)
```
since this is more declarative.

**Comment by rsichny** [comment]
changed

**Comment by pohmelie** [comment]
:+1:

**Pull request merged**